src/main/java/calculator/Tokens.java

```diff
@@ -0,0 +1,75 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import static calculator.Calculator.NUMBER_REGEX;
+import static calculator.Calculator.SIGN_REGEX;
+
+public class Tokens implements Iterable<Token> {
+    private final List<Token> tokens;
+
+    public Tokens(final String[] tokens) {
+        validateExpression(tokens);
+        this.tokens = Arrays.stream(tokens)
+            .map(Token::new)
+            .collect(Collectors.toUnmodifiableList());
+    }
+
+    private static void validateExpression(final String[] tokens) {
```

Comment:
```suggestion
    private void validateExpression(final String[] tokens) {
```
static은 모두 제거되어도 될 것 같아요 :)